server/src/com/cloud/storage/VolumeApiServiceImpl.java

```diff
@@ -1924,6 +1935,23 @@ public Volume migrateVolume(MigrateVolumeCmd cmd) {
                                 throw new InvalidParameterValueException("Cannot migrate ROOT volume of a stopped VM to a storage pool in a different VMware datacenter");
                             }
                         }
+
+                        String rootVolChainInfo = vol.getChainInfo();
+                        if ((vm.getType().equals(VirtualMachine.Type.User)) && (rootVolChainInfo != null) && (!rootVolChainInfo.isEmpty())) {
+                            String rootDiskController = null;
+                            try {
+                                VirtualMachineDiskInfo infoInChain = _gson.fromJson(rootVolChainInfo, VirtualMachineDiskInfo.class);
+                                if (infoInChain != null) {
+                                    rootDiskController = infoInChain.getControllerFromDeviceBusName();
+                                }
+                                UserVmVO userVmVo = _userVmDao.findById(vm.getId());
+                                if ((rootDiskController != null) && (!rootDiskController.isEmpty())) {
+                                    _userVmDao.loadDetails(userVmVo);
+                                    _userVmMgr.persistDeviceBusInfo(userVmVo, rootDiskController);
+                                }
+                            } catch (JsonParseException ignored) {
```

Comment:
At least a warning/debug message should be printed. This approach of silencing exception is never a good idea.